app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CommentsController < ApplicationController
+
+  def create
+    case params[:comment_type]
+    when "Question"
+      comment_target = Question.find(params[:question_id])
+    when "Answer"
+      comment_target = Answer.find(params[:answer_id])
+    end
+    @comment = comment_target.comments.build(comment_params)
+    @comment.user_id = current_user.id
+    if @comment.save
+      redirect_to :back
```

Comment:
You do the same in both branches in which case you could drop the if - but it would be good to show a success / error message to the user.